File: Day18/ContainsDuplicates.java
import java.util.HashSet;

class ContainsDuplicates {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> temp = new HashSet<>();
        for (int num : nums) {
            if (temp.contains(num)) {
                return true;
            }
            temp.add(num);
        }
        return false;
    }

    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 1};
        ContainsDuplicates sol = new ContainsDuplicates();
        System.out.println("Contains Duplicate: " + sol.containsDuplicate(nums)); 
    }
}

File: Day18/ContainsDuplicates.py
class Solution(object):
    def containsDuplicate(self, nums):
        temp = set()
        for num in nums:
            if num in temp:
                return True
            temp.add(num)
        return False

def main():
    nums = [1, 2, 3, 1]
    sol = Solution()
    print("Contains Duplicate:", sol.containsDuplicate(nums)) 
if __name__ == "__main__":
    main()
